homeassistant/components/blueprint/websocket_api.py

```diff
@@ -0,0 +1,98 @@
+"""Websocket API for blueprint."""
+import asyncio
+import logging
+from pathlib import Path
+from typing import Dict, Optional
+
+import async_timeout
+import voluptuous as vol
+
+from homeassistant.components import websocket_api
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import config_validation as cv
+
+from . import importer, models
+from .const import BLUEPRINT_FOLDER, DOMAIN
+
+_LOGGER = logging.getLogger(__package__)
+
+
+@callback
+def async_setup(hass: HomeAssistant):
+    """Set up the websocket API."""
+    websocket_api.async_register_command(hass, ws_list_blueprints)
+    websocket_api.async_register_command(hass, ws_import_blueprint)
+
+
+@websocket_api.async_response
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "blueprint/list",
+    }
+)
+async def ws_list_blueprints(hass, connection, msg):
+    """List available blueprints."""
+    domain_blueprints: Optional[Dict[str, models.DomainBlueprints]] = hass.data.get(
+        DOMAIN
+    )
+
+    if domain_blueprints is None:
+        connection.send_result(msg["id"], {1: 2})
+        return
+
+    blueprint_folder = Path(hass.config.path(BLUEPRINT_FOLDER))
+
+    if not blueprint_folder.exists():
+        connection.send_result(msg["id"], {2: 3})
+        return
+
+    results = {}
+
+    for domain, domain_results in zip(
+        domain_blueprints,
+        await asyncio.gather(
+            *[db.async_get_blueprints() for db in domain_blueprints.values()]
```

Comment:
This will create one executor job per domain. Maybe that's ok?

Reply:
I learn on Supervisor: doing jobs with doing IO on disk - Parallel is bad and make it slower (just hit the disk) / doing IO on network/pipe/memory/process - Parallel gives a real boost (just not hit our system)

Reply:
You're right, that's ugly. We should do it in 1 call.

Reply:
Not going to fix it for MVP, as we only support `automation` right now.